static/app/utils/metrics/useBlockMetric.tsx

```diff
@@ -0,0 +1,76 @@
+import {addErrorMessage, addSuccessMessage} from 'sentry/actionCreators/indicator';
+import {t} from 'sentry/locale';
+import type {MetricMeta, MRI, Project} from 'sentry/types';
+import {getUseCaseFromMRI} from 'sentry/utils/metrics/mri';
+import {useMutation, useQueryClient} from 'sentry/utils/queryClient';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+
+export type BlockOperationType =
+  | 'blockMetric'
+  | 'unblockMetric'
+  | 'blockTags'
+  | 'unblockTags';
+
+type BlockMetricData = [MetricMeta[], unknown, unknown] | undefined;
+
+export const useBlockMetric = (project: Project) => {
+  const api = useApi();
+  const {slug} = useOrganization();
+  const queryClient = useQueryClient();
+
+  const options = {
+    mutationFn: (data: {
+      mri: MRI;
+      operationType: BlockOperationType;
+      tags?: string[];
+    }) => {
+      return api.requestPromise(`/projects/${slug}/${project.slug}/metrics/visibility/`, {
+        method: 'PUT',
+        query: {project: project.id},
+        data: {
+          metricMri: data.mri,
+          project: project.id,
+          tags: data.tags,
+          operationType: data.operationType,
+        },
+      });
+    },
+    onSuccess: data => {
```

Comment:
Should we also trigger a refetch by invalidating the query? To be more defensive?